### PR TITLE
refactor: move `Block` to `evm_rpc_types` 

### PR DESCRIPTION
**Pull request by gregorydemay**
Follow-up on #257  to move the type `Block` to the `evm_rpc_types` crate, so that the public API of `eth_get_block_by_number` only uses types from that crate.

### CODE REVIEW

File: evm_rpc_types/src/response/mod.rs [hunk 1]
```diff
@@ -126,3 +126,88 @@ pub struct TransactionReceipt {
...
+
+    /// Miner
+    pub miner: Hex20,
+
+    /// Mix hash
+    #[serde(rename = "mixHash")]
+    pub mix_hash: Hex32,
+
+    /// Nonce
+    pub nonce: Nat256,
```
Comment (gregorydemay) [inline evm_rpc_types/src/response/mod.rs:170]: According to the [specification](https://github.com/ethereum/execution-apis/blob/1d4f70e84191bb574286fd7cea6c48795bf73e78/src/schemas/block.yaml#L70) this should be 8 bytes encoding as an hex string. I initially added the type `Hex8` (similar to the other `HexN` types); however this is not compatible with the current Candid API which declares `nonce` as a `nat` (whereas `HexN` is a subtype of `text`). We could introduce the type `Nat64` but this seems hardly worth it (this field was used for PoW blocks). WDYT?

**Review by THLO — APPROVED** [review]

File: evm_rpc_types/src/response/mod.rs (hunk 1 above)
Comment (rvanasa) [reply evm_rpc_types/src/response/mod.rs:170]: Using `Nat256` seems like a reasonable way to address this.

**Review by rvanasa — APPROVED** [review]
LGTM; thanks!

**Comment by rvanasa** [comment]
(Pushed changes from `cargo fmt` to pass CI)

**Pull request merged**